lxd/instance/drivers/driver_qemu.go

```diff
@@ -983,8 +990,39 @@ func (d *qemu) Start(stateful bool) error {
 	}
 
 	// Check for stateful.
-	if stateful && !shared.IsTrue(d.expandedConfig["migration.stateful"]) {
-		return fmt.Errorf("Stateful start requires migration.stateful to be set to true")
+	if stateful {
+		if !shared.IsTrue(d.expandedConfig["migration.stateful"]) {
+			return fmt.Errorf("Stateful start requires migration.stateful to be set to true")
+		}
+
+		_, rootDiskDevice, err := d.getRootDiskDevice()
```

Comment:
I think I'd be tempted to break this section into a separate function along with an associated comment explaining why its needed. That will stop the start function becoming more unwieldy. Thanks